File: JavaClient/src/main/java/com/security/crypto/Ciphers/RsaCiphers.java
package com.security.crypto.Ciphers;

import java.security.PrivateKey;
import java.security.PublicKey;


public interface RsaCiphers {
    String RsaEncrypt(PublicKey pubKey, String plainText) throws Exception;

    String RsaDecrypt(PrivateKey privateKey, String EncryptedText) throws Exception;
}

File: SecureUWPClient/SecureUWPClient/Ciphers/Asymmetric/RsaCiphers.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Security.Cryptography.Core;

namespace SecureUWPClient.Ciphers.RSA
{
    public interface RsaCiphers
    {
        Task<String> RsaEncrypt(String publicKey, String plainText);
    }
}
